### PR TITLE
Implement file_operator content

### LINKED ISSUE #288: Update Operator to get contents of commit files
We'd like to have a common method to get the contents of a file in a commit, something like
```
def commit_content(self, commit, filepath)
```
since this is pretty crucial for level development in #238.

`commit` could support taking a [Commit](https://gitpython.readthedocs.io/en/stable/tutorial.html#the-commit-object) object, or a string commit-ish.

`filepath` should support taking [Path](https://docs.python.org/3/library/pathlib.html) objects and string representations of paths.

### PR DESCRIPTION
**Pull request by sahansk2**
Fixes #288 

Replaces #289 with tests for all operating systems. 
https://github.com/sahansk2/git-gud/actions
CircleCI build error because there's no CircleCI `config.yml`

### CODE REVIEW

**Comment by sahansk2** [comment]
@benthayer , this is ready for review.

**Comment by sahansk2** [comment]
@benthayer , the matrix strategy is properly set up, and I added further modifications so that we can see all occurring errors across all OS's, even if the other OS's failed, and even if the previous (relevant) steps failed, since we don't necessarily want to have multiple commits just to fix flake8, which is the test that's run last. 

See here: https://github.com/sahansk2/git-gud/runs/1024141817?check_suite_focus=true
I cancelled the workflow, but notice how even though pytest for ubuntu-latest failed, flake8 was run anyway.

So, this PR is ready for review again.

**Comment by benthayer** [comment]
@sahansk2 Can you update this so that flake8 only runs once? For now, it's fine that the windows test uses bash, but we should fix that at some point. Also can you put all of the testing commits on another branch? I'd like to merge into master with that first, then get to reviewing this PR. Once we merge that branch into master, then it will be appropriate to merge master into this branch and do a review

**Comment by sahansk2** [comment]
Caching of the commit content doesn't actually seem to be necessary. I've added a commit to show this is the case.

**Comment by sahansk2** [comment]
@benthayer This is ready for review. Could you comment on what needs to be done in order for this to be merged?

File: gitgud/test_operator.py [hunk 1]
```diff
@@ -22,6 +26,21 @@ def level():
...
+@pytest.fixture
+def content_level(gg):
+    from gitgud.skills import all_skills
+    gg.load_level(all_skills['intro']['welcome'])
+
+    write_file("root.txt")
+    os.mkdir("dir")
+    write_file("dir/dirfile.txt")
+    os.mkdir("dir/subdir")
+    write_file("dir/subdir/subdirfile.txt")
```
Comment (benthayer) [inline gitgud/test_operator.py:38]: Why do you add `root.txt` and `subdirfile.txt`? It seems like you don't use them

**Review by benthayer — COMMENTED** [review]
I updated names and generally refactored everything. Most of the testing code is the same, but is just shifted around and put in their own methods. 

As for the implementation code, I mostly separated expressions and assigned them to intermediate variables to make things more clear. The biggest structural difference is the `DirectoryContent` object.

If you like the way things are, then I'm okay to merge

File: gitgud/test_operator.py (hunk 1 above)
Comment (sahansk2) [reply gitgud/test_operator.py:38]: I've now added tests for get_commit_content using subdir*.

**Comment by sahansk2** [comment]
This looks good, I just added subdirectory tests since they were unused.

**Comment by benthayer** [comment]
Great, will check it out within a day or two

**Pull request merged**